### ISSUE TITLE
ngOptions mutilating ordering of values in array

### ISSUE THREAD

**Issue by kumarharsh** [angular/angular.js#9465]
If I have an array with 11 or more items, and I write this piece of code:

```js
var years = [
"001",
"002",
...
"011"
];
```

```html
  <select ng-model="model" ng-options="key as year for (key, year) in years"></select>
```

Then the order in the options is this:

```
[
001,
002,
011,
003
...
]
```
*relevant jsfiddle: [http://jsfiddle.net/97h62/44/](http://jsfiddle.net/97h62/44/)*

![screen shot 2014-10-07 at 21 03 37](https://cloud.githubusercontent.com/assets/760112/4544970/68b46206-4e37-11e4-84d8-ed8683935409.png)

**Comment by petebacondarwin** [comment]
This is fixed by #10639 - see http://jsfiddle.net/8gLagt22/ - ngOptions now doesn't attempt to sort the keys of an object at all.

**Comment by Narretz** [comment]
This is fixed in 1.4. If you are still on 1.3.x, you are encouraged to upgrade. Compared to 1.2 -> 1.3 the upgrade is quite painless.:)
